### ISSUE TITLE
RevisionMiddleware doesn't work correctly with StaticLiveServerTestCase

### ISSUE THREAD

**Issue by stebunovd** [etianen/django-reversion#555]
After django-reversion upgrade 1.10.1 -> 2.0.5 all my Selenium tests for Django admin turned red. I did some investigation I found that transactions were actually rolled back instead of being committed. Here is a workaround which worked for me:
```python
class PatchedRevisionMiddleware(RevisionMiddleware):
    """
    Revision middleware which works correctly with StaticLiveServerTestCase.

    For unknown reason sys.exc_info() returns Django Http404 exception
    when called from StaticLiveServerTestCase, even when request was actually
    completed successfully. That's why I handle exceptions explicitly.
    """
    def _close_revision(self, request, exception=None):
        if self in getattr(request, "_revision_middleware", {}):
            exc_info = sys.exc_info() if exception else (None, None, None)
            request._revision_middleware.pop(self).__exit__(*exc_info)

    def process_response(self, request, response):
        self._close_revision(request)
        return response

    def process_exception(self, request, exception):
        self._close_revision(request, exception)
```
Django 1.9.7, Python 2.7.11

P.S. thank you for your work and providing a major upgrade! A couple of annoying bugs have gone away, and this is awesome.

**Comment by etianen** [comment]
Thanks for the report, and for a suggested fix!

I've pushed a commit, based on your solution, to master. Please try it out, and let me know if it works for you.

**Comment by stebunovd** [comment]
switched to master branch, it works for me. Thanks!

**Comment by schinckel** [comment]
For what it's worth, this bug affects more than just the test indicated above: it actually breaks _all_ saving of data when running under gunicorn.

Gunicorn must throw and then handle some exception, which remains in `sys.exc_info()`, resulting in a `ROLLBACK` being emitted by the middleware at this point.

Ouch.

**Comment by etianen** [comment]
Fixed in v2.0.6
